torchvision/io/video.py

```diff
@@ -153,14 +153,11 @@ def _read_from_stream(
         gc.collect()
 
     if pts_unit == "sec":
+        # TODO: we should change all of this from ground up to simply take
+        # sec and convert to MS in C++ 
         start_offset = int(math.floor(start_offset * (1 / stream.time_base)))
         if end_offset != float("inf"):
             end_offset = int(math.ceil(end_offset * (1 / stream.time_base)))
-    else:
-        warnings.warn(
-            "The pts_unit 'pts' gives wrong results and will be removed in a "
-            + "follow-up version. Please use pts_unit 'sec'."
-        )
```

Comment:
Instead of this warning, which seems to do the right thing, can't we change the reference script to not use `"pts"`?

Reply:
Not really; or rather, it used to. The issue is that the backend (both pyav and VideoReader) consume data in `pts` which is unfortunate as they are not standardised per stream (e.g. audio and video have different timebases so pts=24 means different things). In a PR some time ago, we've tried to fix this by introducing `sec` as a default unit that would then be converted to appropriate stream. This caused an imprecision issue (bc thigs would be incorrectly rounded) which broke all datasets bc VideoClips creation requires fairly precise pts and such rounding erros cause major issues. We've initially fixed these by using `pts` as a default in VideoClips (which raises this warrning). This would be fine (i.e. I'd keep the warning and try to re-work the video clips) if all else was equal, but the PR mentioned above was reverted for internal issues, so this warrning is not really relevant anymore. 

Having said that, I have a major PR with a VideoReading re-work in plan that will make this redundant in a way.

Reply:
@prabhat00155  most likely, as he was the one who originally implemented and then reverted the sync PR.
But we can also keep this open as I'm bound to finish the rework by the end of next week

Reply:
As discussed with @datumbox 


We'll put this warning back. There should only be 2 reasons to remove this warning:

- fix the actual bug
- completely forbid "pts".